altos-rust/altos-core/libs/heap/free_list_allocator/src/free_list.rs

```diff
@@ -196,13 +182,12 @@ impl FreeList {
             return;
         }
         let (mut previous, current) = {
-            let free_mem_between = |previous, current| {
+            self.find_block(|previous, current| {
                 if alloc_block_ptr == current {
-                    panic!("attempt to free memory that's already free");
+                    panic!("deallocate - attempt to free memory that's already free");
                 }
                 previous < alloc_block_ptr && alloc_block_ptr < current
-            };
-            self.find_block(&free_mem_between)
+            })
         };
```

Comment:
We can remove this enclosing scope now that the assignment is just one statement.